docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+This is a Kubernetes controller that reconciles Kubewarden's Custom Resources.
+This controller creates parts of the Kubewarden stack.
+It also translates Kubewarden configuration into Kubernetes directives.
 
-- Kubewarden policies: these are WebAssembly modules that hold the validation
-  or mutation logic. These are covered in depth inside of [this chapter](/writing-policies/index.md).
+- Kubewarden policies:
+These are WebAssembly modules holding the validation or mutation logic.
+WebAssembly modules have detailed documentation in the
+[writing policies](/writing-policies/index.md) sections.
 
 - [`policy-server`](https://github.com/kubewarden/policy-server):
-  this component receives the requests to be validated. It does that
-  by executing Kubewarden's policies.
+The `policy-server` receives requests for validation.
+It validates the requests by executing Kubewarden policies.
 
-At the bottom of the stack, Kubewarden's integrates with Kubernetes using the
-concept of [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+Kubewarden integrates with Kubernetes using
+[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
 In particular, Kubewarden operates as a Kubernetes Admission Webhook.
-`policy-server` is the actual Webhook endpoint that is reached by Kubernetes
-API server to validate relevant requests.
+The `policy-server` is the Webhook endpoint called by the Kubernetes API server to validate requests.
 
-Kubernetes is made aware of the existence of Kubewarden's Webhook endpoints by
-`kubewarden-controller`. This is done by registering either
-a `MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration`
-object.
+<!--TODO:Have I got this next paragraph right?-->
 
-This diagram shows the full architecture overview of a cluster running
-the Kubewarden stack:
+Kubernetes discovers Kubewarden's Webhook endpoints using `kubewarden-controller`.
+This works by Kubewarden registering either a `MutatingWebhookConfiguration`
+or a `ValidatingWebhookConfiguration` object with Kubernetes.
+
+<!--TODO:@vicuad points out we need a paragraph about the Audit Scanner.-->
+
+The diagram shows the architecture of a cluster running the Kubewarden stack:
+
+<!--TODO:To be clear. The stack refers to the entire diagram rather that the
+stacked components for e.g., CAP.-->
 
 ![Full architecture](/img/architecture.png)
 
-## Journey of a Kubewarden policy
+## The journey of a Kubewarden policy
+
+The architecture diagram appears complex at first. This section covers it step by step.
+
+### Default policy server
 
-The architecture diagram from above can be intimidating at first, this
-section explains it step by step.
+On a new cluster, the Kubewarden components defined are:
 
-### Default Policy Server
+<!--TODO:How are CRDs shown in the diagram-->
```

Comment:
Yes, CRDs are just pushed to the k8s apiserver so it gains awareness of `PolicyServer`, `ClusterAdmissionPolicies`, `AdmissionPolicies`, and we can instantiate them. Since the diagram has some of those instantiated, I think it's clear that they are needed.